### ISSUE TITLE
BUG:  Network requests always fails

### ISSUE THREAD

**Issue by joelewis** [stackblitz/webcontainer-core#142]
**Network requests are not handled**
Any library that involves making http(s) requests (axios / request / etc) are erroring out when making actual requests.

**Link to the blitz that caused the error**
https://stackblitz.com/edit/node-zgjiqf

**Steps to reproduce:**
1. Get to the terminal
2. Install `npm install --save request` which works well
3. Use the request library's API in index.js (as indicated in the example blitz)

**Expected behavior:**
Expected the network request to go through or at least fail with a relevant error message

**Version of webcontainer:**
Hash: 12423d4ada17c5bf9dcd8d2c2bd877a5f9b12d08

**Desktop (please complete the following information):**
Browser name  = Chrome 
Full version  = 90.0.4430.212
Major version = 90
navigator.appName = Netscape
navigator.userAgent = Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36
performance.memory = {
  "totalJSHeapSize": 109764941,
  "usedJSHeapSize": 105507769,
  "jsHeapSizeLimit": 4294705152
}

**Comment by SamVerschueren** [comment]
Hi @joelewis, thanks for trying it out.

So the reason it fails here is because of CORS. We can't magically circumvent the browsers security sandbox. For instance, if you use the Pokemon API, you will see that it works https://stackblitz.com/edit/node-6yu9qs. We already discussed internally if we could do something about this, but it won't happen soon I guess.

**Comment by d3lm** [comment]
Yes, precisely what @SamVerschueren said. It really depends on the endpoint you are trying to request if it has CORS enabled and what origins, headers, methods are allowed. If the server rejects certain origins then this will cause the request to fail due to CORS. There are solutions to this problem and we have already discussed some of them but as Sam said, this won't happen soon and might take a bit for us to get right. There are several things to consider and scaleability plays a big role in this.

**Comment by joelewis** [comment]
Got it. Thanks! 
Curious: What solutions came into the discussion to this problem? 
A couple of what comes to my mind: 
1. Can't we run a proxy server that routes any request from the client to go through the proxy, thereby simulating cors-less nodejs env? 
2. Maybe a stackblitz browser extension that proxies those requests from an extension environment, which is less constrained.

**Comment by SamVerschueren** [comment]
Thanks for thinking with us @joelewis!

Solution number 1 is what we thought about as well. But that would involve a proxy which will cost a lot of 💸  probably because every request has to go through that proxy.

I don't think we thought about an extension. Not sure if that would work but that might be a good idea. Although we want users to have as less friction as possible without installing anything.

So option number 1 is I guess from a user point of view definitely the best option here, because it will just run out of the box.

**Comment by d3lm** [comment]
An extension was already on the table and in fact we actually had one for a while but it's just not a great experience. Users have to install and not as seamless as a proxy. So a proper solution to this problem is definitely a proxy but there are several things to keep in mind which is scalability, costs or availability.

**Comment by aleclarson** [comment]
+1 this is a big issue for [viteflare](https://github.com/alloc/viteflare) compatibility with stackblitz

I think a browser extension is the right solution. Stackblitz could detect CORS errors and offer to install the proxy extension "just in time". The act of installing an extension is trivial for users, and much better of an experience than having no recourse in the face of CORS errors. :)

**Comment by d3lm** [comment]
@aleclarson No, unfortunately you can't catch CORS errors. But we can definitely intercept CORS requests.

**Comment by d3lm** [comment]
Hey, just wanted to drop a message and say that we are still aware of CORS issues, but I don't think that there will be a solution on our end any time soon and I suggest to work around CORS issues yourself by using a CORS proxy. You could use a Cloudflare Worker for that.
